elementor/elementor
# Reduce-motion accessibility support for CSS animation library

## PR Checklist



## PR Type
What kind of change does this PR introduce?


## Summary

This PR can be summarized in the following changelog entry:

* Add reduce-motion accessibility support for CSS animation library

## Description
An explanation of what is done in this PR

A new media query selector has been introduced to allow users who suffer motion sickness / vertigo to disable UI animations (via System settings). The media query allows website developers to extend this preference into their site design, providing an option to disable excessive or non-critical animation to users who desire it. 

Having this option built into Elementor will be a great benefit to accessibility compliance, and give each  website visitor greater control over their individual experience.

For Elementor, I have proposed using the media query selector on the CSS provided by the animate.css library. I have targeted the .animated class to cover as many animated elements as possible (eg entrance animations & popup animations). The code disables animation for anyone with "reduce motion" set in their system settings, but will default to normal behaviour for users who don't toggle this option, or explicitly set it to enable animation.

Reduce-motion currently has support on macOS / iOS / Android + Chrome & Safari, however it is expected to be adopted by more platforms and browsers in the future.

More information:
https://css-tricks.com/introduction-reduced-motion-media-query/
https://developers.google.com/web/updates/2019/03/prefers-reduced-motion
https://drafts.csswg.org/mediaqueries-5/#prefers-reduced-motion

Code changes tested against release/v2.5.14 branch

## Test instructions
This PR can be tested by following these steps:

* Compile the SCSS in this PR to generate CSS build assets
* Create a test page in Elementor and apply an Entrance animation to an element
* Preview the page and ensure the animation plays
* Enable 'reduce motion' preferences (in macOS Mojave, go to System Preferences -> Accessibility -> Display -> check 'reduce motion'
* The new setting will immediately take effect - reload the test page and the animation should not play, and a static element should only display

## Quality assurance


Fixes #
